### PR TITLE
add callback options to SignerPlugin

### PR DESCRIPTION
**Pull request by ququzone**
**What type of PR is this?**

> /kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

#300

### CODE REVIEW

File: src/action/method.ts [hunk 1]
```diff
@@ -3,12 +3,18 @@ import { IAction, IRpcMethod } from "../rpc-method/types";
...
 export interface SignerPlugin {
-  signAndSend?(envelop: Envelop): Promise<string>;
+  signAndSend?(envelop: Envelop, options?: PluginOpts): Promise<string>;
 
-  signOnly?(envelop: Envelop): Promise<SealedEnvelop>;
+  signOnly?(envelop: Envelop, options?: PluginOpts): Promise<SealedEnvelop>;
 
   getAccount?(address: string): Promise<Account>;
+
+  getAccounts?(): Array<string>;
```
Comment (puncsky) [inline src/action/method.ts]: probably return Promise.

File: src/action/method.ts [hunk 2]
```diff
@@ -3,12 +3,18 @@ import { IAction, IRpcMethod } from "../rpc-method/types";
 import { Envelop, SealedEnvelop } from "./envelop";
 import { ClaimFromRewardingFund, Execution, Transfer } from "./types";
 
+export class PluginOpts {
```
Comment (puncsky) [inline src/action/method.ts]: should probably be similar to MethodExecuteParameter

```
  account?: IAccount; // account
  from?: string; // address
  amount?: string;
  gasLimit?: string;
  gasPrice?: string;
```

File: src/action/method.ts (hunk 1 above)
Comment (ququzone) [reply src/action/method.ts]: resolved

File: src/action/method.ts (hunk 2 above)
Comment (puncsky) [reply src/action/method.ts]: use interface?

File: src/action/method.ts [hunk 3]
```diff
@@ -3,12 +3,18 @@ import { IAction, IRpcMethod } from "../rpc-method/types";
...
 export interface SignerPlugin {
-  signAndSend?(envelop: Envelop): Promise<string>;
+  signAndSend?(envelop: Envelop, options?: PluginOpts): Promise<string>;
 
-  signOnly?(envelop: Envelop): Promise<SealedEnvelop>;
+  signOnly?(envelop: Envelop, options?: PluginOpts): Promise<SealedEnvelop>;
 
   getAccount?(address: string): Promise<Account>;
+
+  getAccounts?(): Promise<Array<string>>;
```
Comment (puncsky) [inline src/action/method.ts]: probably use 

```
getAddresses?(): Promise<Array<string>>;
```

or 

```
getAccounts?(): Promise<Array<Account>>;
```

probably the second one

File: src/__test__/iotx.plugin.test.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+          envelop
+        );
+      }
+    }
+  });
+
+  await sleepPromise(3000);
+
+  const hash = await antenna.iotx.sendTransfer({
+    from: TEST_ACCOUNT.address,
```
Comment (puncsky) [inline src/__test__/iotx.plugin.test.ts]: something like `from: antenna.iotx.accounts[0]`

**Pull request merged**